main/plugins/service.ts

```diff
@@ -0,0 +1,27 @@
+
+import {Format} from '../common/types';
+import {Schema} from '../utils/ajv';
+import {ShareServiceContext} from './service-context';
+
+export interface Service<Config = any> {
+  title: string;
+  configDescription?: string;
+  config?: {[P in keyof Config]: Schema};
+}
+
+export interface ShareService<Config = any> extends Service<Config> {
+  formats: Format[];
+  action: (context: ShareServiceContext) => void;
+}
+
+export interface EditService<Config = any> extends Service<Config> {
+  action: () => {};
+}
+
+export interface RecordService<Config = any> extends Service<Config> {
+  willStartRecording?: () => {};
+  didStartRecording?: () => {};
+  didStopRecording?: () => {};
+  willEnable?: () => {};
+  cleanUp?: () => {};
+}
```

Comment:
We should find a way (when this is done) to export this somehow as a package so plugins written in TS can leverage these types